### PR TITLE
Retry audio loop when the output audio device is unplugged or switched

### PR DESCRIPTION
**Pull request by algologarithm**
When the audio output device switched or unplugged on the PC which run the ALVR server,the streaming would be cut off,and then sometimes the alvr in the HMD crashed at the same time. This modify can retry the game_audio_loop in order to avoid this situation.

### CODE REVIEW

File: alvr/audio/src/lib.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ pub struct AudioDevice {
 impl AudioDevice {
     pub fn new(
         linux_backend: Option<LinuxAudioBackend>,
-        id: AudioDeviceId,
+        id: &AudioDeviceId,
```
Comment (zarik5) [inline alvr/audio/src/lib.rs:60]: Why this change? If needed you can clone `AudioDeviceId`.

File: alvr/server/src/connection.rs [hunk 2]
```diff
@@ -625,61 +625,80 @@ async fn connection_pipeline() -> StrResult {
...
+                        warn!("New audio device Failed : {e}");
+                        time::sleep(CONTROL_CONNECT_RETRY_PAUSE).await;
+                        continue;
+                    }
+                };
+                let mute_when_streaming = desc.mute_when_streaming;
 
+                #[cfg(windows)]
                 unsafe {
+                    let device_id = match alvr_audio::get_windows_device_id(&device) {
```
Comment (zarik5) [inline alvr/server/src/connection.rs:648]: How likely is this to fail? Should you bail out with `?` instead? I feel this is the whole point of the PR but I'm asking just to make sure. The same applies to the following code.

File: alvr/server/src/connection.rs [hunk 3]
```diff
@@ -625,61 +625,80 @@ async fn connection_pipeline() -> StrResult {
...
+                            OpenvrPropValue::String(device_id),
                         ),
                     )
                 }
-            }
+                let new_sender = sender.clone();
+                match alvr_audio::record_audio_loop(device, 2, mute_when_streaming, new_sender)
+                    .await
+                {
+                    Ok(_) => warn!("Audio Normal exit."),
```
Comment (zarik5) [inline alvr/server/src/connection.rs:664]: Do no emit logging in the "happy" path. Unless you want to use `debug!()` but I don't think it's necessary.

**Review by zarik5 — CHANGES_REQUESTED** [review]
Thank you. I'm not able to test now, maybe @korejan can.

File: alvr/audio/src/lib.rs (hunk 1 above)
Comment (zarik5) [reply alvr/audio/src/lib.rs:60]: Nevermind, I think It's reasonable. `AudioDeviceId` may allocate a string and it seems the method does not need to consume the value so taking a reference is justified.

**Comment by zarik5** [comment]
If you need to make further changes you don't need to close the PR and make a new one. Just push to the same branch.

File: alvr/server/src/connection.rs (hunk 2 above)
Comment (algologarithm) [reply alvr/server/src/connection.rs:648]: Yes, because I want the "loop" make sense, I can't use ? in the loop , so according to rust syntax, I must deal with every Result<T,E> instead of using ?.

Comment (zarik5) [reply alvr/server/src/connection.rs:648]: I see. We can keep this code as it is.

**Comment by algologarithm** [comment]
@zarik5 Excuse me ,I want to know if this commit has been tested ,and is there any problem in the code ?

**Comment by zarik5** [comment]
I don't have the capacity to properly test this. I'll merge and people can report if this breaks anything.
